### ISSUE TITLE
Use TaxRates instead of tax_percent

### ISSUE THREAD

**Issue by kennethsinder** [laravel/cashier#657]
With the [recent ship](https://stripe.com/docs/billing/subscriptions/taxes) of EU invoicing and `TaxRate`s by Stripe Billing as the new way to model taxes, it might be a good first step to update `syncTaxPercentage()` in `Subscription` to create `TaxRate` objects instead of setting it via the [deprecated](https://stripe.com/docs/api/subscriptions/create#create_subscription-tax_percent) `tax_percent`.

This would involve:
* Bumping the pinned version of stripe-php to 6.34.0 (https://github.com/stripe/stripe-php/commit/564b93b3a760af6628832fbccd6aeab55f993c2c) so that `TaxRate`s are supported.
* Changing this method to create a [`TaxRate`](https://stripe.com/docs/api/tax_rates/object) first via the Stripe API and then apply the resulting tax rate ID to the `Subscription`: https://github.com/laravel/cashier/blob/707ef5ec28974f74c57461ef25c125c46050d885/src/Subscription.php#L500 For now, we could hard-code the tax `display_name` to `"Tax"` when we create that tax rate to get approximately the same functionality that now happens implicitly when setting a `tax_percent`.

There are no direct immediate benefits of doing this, but it unlocks:
* In a separate proposal + PR, we could have multiple `TaxRate`s, at the subscription and subscription item level, to model items that are taxed differently on the same subscription.
* We could support `inclusive` taxes, find a way to support passing in a customer's `jurisdiction` and set that so it can be used for reporting, etc.

This can be done separately from Payment Intent / SCA support which is already listed as a separate issue (https://github.com/laravel/cashier/issues/636) in this repo, but it seems like a modest step that can unlock richer tax functionality.

**Comment by driesvints** [comment]
Heya, thanks for the detailed issue! I'll be happy to look into this further but only after we've made all the changes for the upcoming v10 release. We might be able to get this in that release as well but for now I'd like to focus on getting payment intents support into Cashier.

**Comment by kennethsinder** [comment]
Awesome, sounds good! I should have clarified I'm offering to try implementing this myself but it sounds like you're looking to finalize v10 first, but if it becomes a good time for me to jump in, let me know!

**Comment by lucidlive** [comment]
Would greatly appreciate this feature!

**Comment by ReckeDJ** [comment]
Awesome. This is definitely something that we need to operate in different countries in the EU (and outside the EU later). Maybe I can help in some way, just let me know.

**Comment by azatotht** [comment]
This would be great indeed. We need to be able to charge 2 different taxes in Stripe (Canadian federal and provincial taxes) which both have different rates. If I understand well, we need this update in order to be able to do that in the new Stripe API, right now we can only have one tax amount.

**Comment by pierrocknroll** [comment]
I think this could be taken into account now since Cashier 10 is out

**Comment by johnRivs** [comment]
Is the new Tax Rate the only way to apply an inclusive tax percentage to a new subscription? Currently, every subscription I start with Cashier is creating and reusing an exclusive tax rate instead of using the inclusive one I have set up.

If this proposal would help with this, that'd be awesome.

**Comment by lsmith77** [comment]
@kennethsinder did you ever turn your very concrete suggestions into a PR?

**Comment by pierrocknroll** [comment]
Hi, 
I try to make a PR but I'm stuck...

When I want to add a default_tax_rates key to the buildPayload method of the SubscriptionBuilder, I have this error : "Received unknown parameter: default_tax_rates"
I don't understand why since this parameter is in the documentation...
My API version in the Stripe Dashboard is up-to-date and the const STRIPE_VERSION is equal to '2019-11-05'.

But if I update the subscription after its creation with
```php
$stripeSubscription->default_tax_rates = ['foo'];
$stripeSubscription->save();
```
it works...

Maybe somebody here can help me because I thought my PR was ready...

**Comment by pierrocknroll** [comment]
I created a small PR #816. With it, you can add a taxRates method in your User model that returns an array of TaxRates ids to add to the Subscription creation.

**Comment by kennethsinder** [comment]
> @kennethsinder did you ever turn your very concrete suggestions into a PR?

Hi @lsmith77, nope I didn't get a chance to yet and I didn't get the go-ahead from any of the repo maintainers. I'm new to open source so I didn't want to code anything up before hearing back. Sounds like @pierrocknroll, @driesvints and Taylor Otwell have it under control in #818 😄

**Comment by driesvints** [comment]
I'm gonna start working on this tomorrow.

**Comment by driesvints** [comment]
WIP pr here: https://github.com/laravel/cashier/pull/830

**Comment by driesvints** [comment]
I just finished up the PR for this. I'd really appreciate any feedback before it gets merged. Please comment on the PR.

https://github.com/laravel/cashier/pull/830
